components/ingest-service/backend/elastic/actions.go

```diff
@@ -31,6 +31,43 @@ func (es *Backend) CreateBulkActionRequest(action backend.InternalChefAction) el
 	return es.createBulkUpdateRequestToTimeseriesIndex(mapping, time, action)
 }
 
+func (es *Backend) DeleteActionProjectTag(ctx context.Context,
+	projectTagToBeDelete string) (string, error) {
+	script := `
+		if (ctx._source['projects'] != null) {
+			def foundIndex = -1;
+			for (def index = 0; index < ctx._source.projects.size(); index++) {
+				if (ctx._source.projects[index] == params.project) {
+					foundIndex = index;
+					break;
+				}
+			}
+
+			if (foundIndex != -1) {
+				ctx._source.projects.remove(foundIndex);
+			}
+		}
```

Comment:
`ctx._source.projects.remove(projectTagToBeDelete)` does not work because the `remove` function only takes an index of the item to remove.

Reply:
😢